internal/pkg/agent/application/managed_mode.go

```diff
@@ -209,12 +212,17 @@ func (m *managedConfigManager) Run(ctx context.Context) error {
 
 	// pass actions collected from gateway to dispatcher
 	go func() {
+		t := time.NewTimer(dispatchFlushInterval)
 		for {
 			select {
 			case <-ctx.Done():
 				return
+			case <-t.C: // periodically call the dispatcher to handle scheduled actions.
```

Comment:
You need to add a test for this, preferably one that catches the original bug you are fixing here.